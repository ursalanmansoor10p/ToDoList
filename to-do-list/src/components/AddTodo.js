import React from "react";
import { db } from "../firebase"
import { collection, addDoc} from "firebase/firestore"

export default function AddTodo() {
    const [title, setTitle] = React.useState("");

        const handleSubmit = async (e) => {
        //e.preventDefault(); prevents any reload when submitting
        //form 
        e.preventDefault();
        if(title !== ""){
            await addDoc(collection(db,'todos'), {
               // Data should be passed as object
                title,
                completed: false,
            });
            setTitle("");
          }
    }

    return (
        <div data-testid="add-todo" >
        <form name="add-form" onSubmit={handleSubmit}>
          <div className="input_container">
            <input
              type="text"
              placeholder="Please enter text"
              value={title}
              onChange={(e) => setTitle(e.target.value)}
              className="input_style"
            />
            <div className="btn_container">
            <button className="button button1">+</button>
             </div>

          </div>
          
        </form>
      </div>
    );


}