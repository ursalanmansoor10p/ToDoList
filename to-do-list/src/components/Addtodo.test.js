import React from "react";
import "@testing-library/jest-dom";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import AddTodo from "./AddTodo";
import { addDoc } from "firebase/firestore";

jest.mock("firebase/firestore", () => ({
  collection: jest.fn(),
  addDoc: jest.fn(),
}));
jest.mock("../firebase", () => ({
  db: jest.fn(),
}));

describe("AddTodo Component", () => {
  test("should match snapshot", () => {
    const { container } = render(<AddTodo />);
    expect(container).toMatchSnapshot();
  });

  test("Button is rendered", () => {
    render(<AddTodo />);
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("Textbox is rendered", () => {
    render(<AddTodo />);
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  test("should add document in firebase on submit", async () => {
    render(<AddTodo />);
    const textBox = screen.getByRole("textbox");
    fireEvent.input(textBox, { target: { value: "abc" } });
    expect(textBox.value).toBe("abc");
    const form = screen.getByRole("form");
    fireEvent.submit(form);
    expect(addDoc).toBeCalled();
    await waitFor(() => expect(textBox.value).toBe(""));
  });
});
