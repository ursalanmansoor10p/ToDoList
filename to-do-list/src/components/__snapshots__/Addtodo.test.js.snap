// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo Component should match snapshot 1`] = `
<div>
  <div
    data-testid="add-todo"
  >
    <form
      name="add-form"
    >
      <div
        class="input_container"
      >
        <input
          class="input_style"
          placeholder="Please enter text"
          type="text"
          value=""
        />
        <div
          class="btn_container"
        >
          <button
            class="button button1"
          >
            +
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
